src/common/core/core.js

```diff
@@ -1961,6 +1961,26 @@ define([
          * @func
          */
         this.getLibraryMetaNodes = core.getLibraryMetaNodes;
+
+        /**
+         * The function traverses the sub-tree of the project starting with the given root and calls the
+         * visit function for every node.
+         *
+         * @param {module:Core~Node} root - the root node of the sub-tree that needs to be traversed.
+         * @param {object} options - parameters to control the traversing.
+         * @param {bool} [options.excludeRoot = false] - controls whether the root should be excluded from visit.
+         * @param {integer} [options.maxParallelLoad = 10]- the maximum number of parallel loads allowed.
+         * @param {bool} [options.blockingVisit = true] - controls if the loading should wait for the visit
+         * function to return.
+         * @param {integer} [options.speed = 10] - the frequency in milliseconds
+         * to check for the next task in the traverse.
+         * @param {function()} visitFn - the synchronous visitation function that will be called for
```

Comment:
Technically we cannot limit that, we just expect the visit function to be synchronous.
Majority of the exchange formats can be handled in a synchronous way (in regards to the visit function) and we can extend it if there is a real need (by adding another version of the traverse that deals with an asynchronous visit function).
The function footprint is less complex this way.